watchman/decorators.py

```diff
@@ -23,8 +34,19 @@ def wrapped(*args, **kwargs):
             }
             # The check contains several individual checks (e.g., one per
             # database). Preface the results by name.
-            if args:
-                response = {args[0]: response}
+            if arg_name:
+                response = {arg_name: response}
+                logger.exception(
+                    "Error calling '%s' for '%s': %s",
+                    check_name,
+                    arg_name,
+                    e.__class__.__name__
+                )
+            else:
+                logger.exception(
+                    "Error calling '%s': %s", check_name, e.__class__.__name__
```

Comment:
Can you provide an example of what the output of this log would look like? `e.__class__.__name__` would give the "main" error message? Should we be including the full stack trace as well?

Reply:
+1 to capturing the actual message from the exception instead of the class name

Using `logger.exception` implies `exc_info=True` which would capture exception information to be added to the logging output.

Reply:
Previously, there was just the class name of the exception, not the actual message. I have updated it to include the class name and the message, so overall it looks like:

`Error calling '_check_database' for 'default': ValueError('Some value error')`

As Robert mentions, since we're using `logger.exception`, it will automatically include the whole stack trace in the message.